pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -309,18 +310,18 @@ func (OutboundProxyGenerator) generateEDS(
 	return resources, nil
 }
 
-// InferProtocol infers protocol for the destination listener. It will only return HTTP when all endpoints are tagged with HTTP.
-// Deprecated: for new policies use InferProtocol from pkg/plugins/policies/xds/clusters.go
-func InferProtocol(proxy *model.Proxy, clusters []envoy_common.Cluster) core_mesh.Protocol {
-	var allEndpoints []model.Endpoint
+func InferProtocol(servicesProtocol map[string]core_mesh.Protocol, clusters []envoy_common.Cluster) core_mesh.Protocol {
+	var protocol core_mesh.Protocol = core_mesh.ProtocolUnknown
 	for _, cluster := range clusters {
 		serviceName := cluster.Tags()[mesh_proto.ServiceTag]
-		endpoints := model.EndpointList(proxy.Routing.OutboundTargets[serviceName])
-		allEndpoints = append(allEndpoints, endpoints...)
-		endpoints = proxy.Routing.ExternalServiceOutboundTargets[serviceName]
-		allEndpoints = append(allEndpoints, endpoints...)
+		currentProtocol := servicesProtocol[serviceName]
+		if protocol == core_mesh.ProtocolUnknown && currentProtocol != "" {
```

Comment:
Isn't this handled in `GetCommonProtocol` ?

Reply:
It isn't but can add it